modules/new_profile/php/profile.class.inc

```diff
@@ -0,0 +1,176 @@
+
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "application/json")
+                ->withStatus(201)
+                ->withHeader("Allow", "POST")
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode($result)
+                    )
+                );
+        break;
+        case "GET":
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "application/json")
+                ->withStatus(201)
+                ->withHeader("Allow", "GET")
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode($this->getDefaultFields())
+                    )
+                );
+            break;
+        default:
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(405)
+                ->withHeader("Allow", "GET,POST");
+        }
+
+    }
+    /**
+     * This function will return an array
+     * This array will help to initializing react form
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    function getDefaultFields(): array
+    {
+        $DB     = \Database::singleton();
+        $config = \NDB_Config::singleton();
+        $user   = \User::singleton();
+        $result['startYear']  = $config->getSetting('startYear');
+        $result['endYear']    = $config->getSetting('endYear');
+        $result['ageMax']     = $config->getSetting('ageMax');
+        $result['ageMin']     = $config->getSetting('ageMin');
+        $result['dobFormat']  = $config->getSetting('dobFormat');
+        $result['edc']        = $config->getSetting('useEDC');
+        $result['useProject'] = $config->getSetting('useProjects');
+        $result['sex']        = array(
+                                 'male'   => 'Male',
+                                 'female' => 'Female',
+                                );
+        $result['pscidSet']   = "false";
+        //get sites for the select dropdown
+        $user_list_of_sites = $user->getData('CenterIDs');
+        $num_sites          = count($user_list_of_sites);
+        if ($num_sites >=1) {
+            foreach ($user_list_of_sites as $key => $siteID) {
+                $center = $DB->pselectRow(
+                    "SELECT CenterID as ID, Name FROM psc WHERE CenterID =:cid",
+                    array('cid' => $siteID)
+                );
+                $psc_labelOptions[$siteID] = $center['Name'];
+            }
+        } else {
+            $psc_labelOptions = array(null => '');
+        }
+         $result['site'] = $psc_labelOptions;
+         //get projects for the select dropdown
+        $projects = \Utility::getProjectList();
+        foreach ($projects as $key=>$value) {
+             $list_of_projects[$key] = $value;
+        }
+        $result['project'] = $list_of_projects;
```

Comment:
@kongtiaowang Line number 167 needs to be:
`$result['project'] = isset($list_of_projects) ?? null;`
and otherwise the frontend will display "An error occured while loading the page." because $list_of_projects can be undefined on a fresh install of LORIS.

Reply:
Should actually be 
```php
$result['project'] = $list_of_projects ?? null;
```

isset() is implicit in `??`.

Reply:
@kongtiaowang please use the code from @johnsaigle because the isset isn't necessary.